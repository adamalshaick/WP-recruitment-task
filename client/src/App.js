import React, { lazy, Suspense } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Loading from "./components/common/Loading";
const Landing = lazy(() => import("./components/landing/Landing"));
const NotFound = lazy(() => import("./components/common/NotFound"));

export const App = () => {
  return (
    <Router>
      <Suspense fallback={<Loading />}>
        <Switch>
          <Route exact patch="/" component={Landing} />
          <Route component={NotFound} />
        </Switch>
      </Suspense>
    </Router>
  );
};

export default App;
