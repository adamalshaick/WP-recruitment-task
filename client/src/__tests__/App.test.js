import React from "react";
import App from "../App";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const wrapper = shallow(<App />);
describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
