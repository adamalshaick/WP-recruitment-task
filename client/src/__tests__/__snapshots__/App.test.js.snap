// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<BrowserRouter>
  <React.Suspense
    fallback={<Loading />}
  >
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
      />
    </Switch>
  </React.Suspense>
</BrowserRouter>
`;
